### ISSUE TITLE
ResonanceMolSupplier potentially stuck in infinite loop

### ISSUE THREAD

**Issue by eguidotti** [rdkit/rdkit#6201]
**Describe the bug**
Chem.ResonanceMolSupplier hangs (enters an infinite loop?) when applied to `CCN(CC)c1ccc2nc3ccc(N(CC)CC)cc3[o+]c2c1`

This is the same of https://github.com/rdkit/rdkit/issues/5406 which was supposed to be fixed in a previous release. The issue is still there in rdkit version `2022.09.5`

**To Reproduce**
```python
from rdkit import Chem
smi = 'CCN(CC)c1ccc2nc3ccc(N(CC)CC)cc3[o+]c2c1'
mol = Chem.MolFromSmiles(smi)
suppl = Chem.ResonanceMolSupplier(mol)
suppl[0] # -> endless
```

**Expected behavior**
Same as https://github.com/rdkit/rdkit/issues/5406

**Configuration (please complete the following information):**
 - RDKit version: `2022.09.5`
 - OS: MacOS Monterey
 - Python version (if relevant): 3.9
 - Are you using conda? no
 - If you are using conda, which channel did you install the rdkit from? pypi
 - If you are not using conda: how did you install the RDKit? pip install rdkit-pypi

**Additional context**

**Comment by ptosco** [comment]
@eguidotti I can't reproduce this on current `master`, so I guess for some reason that bug fix didn't make the `2022.09.5` patch release:
```
from rdkit import Chem
from rdkit.Chem.Draw import MolsToGridImage

smi = 'CCN(CC)c1ccc2nc3ccc(N(CC)CC)cc3[o+]c2c1'
mol = Chem.MolFromSmiles(smi)
suppl = Chem.ResonanceMolSupplier(mol)

MolsToGridImage(suppl)
```
![image](https://user-images.githubusercontent.com/5244385/225884960-bbe3297e-2c2e-40d4-a295-da009f9c3625.png)

I am closing this bug as it does not seem to affect current RDKit `master` with the molecule in your report, please feel free to reopen it if you find an occurrence of the bug with another molecule.

**Comment by eguidotti** [comment]
Thanks for looking into this! I'll simply wait for the next release then 👍
